test/integration/servicebus/eventhubs/client.test.js

```diff
@@ -0,0 +1,215 @@
+'use strict';
+var AMQPClient = require('../../../../lib/index.js').Client,
+  Policy = require('../../../../lib/index').Policy,
+  Message = require('../../../../lib/types/message'),
+  Promise = require('bluebird'),
+  config = require('./config'),
+  expect = require('chai').expect,
+  uuid = require('uuid'),
+  debug = require('debug')('amqp10:test:servicebus:queues'),
+  _ = require('lodash'),
+
+  M = require('../../../../lib/types/message');
+
+var test = {};
+describe('ServiceBus', function() {
+  describe('EventHubs', function() {
+
+    beforeEach(function() {
+      if (!!test.client) test.client = undefined;
+      test.client = new AMQPClient(Policy.ServiceBusQueue);
+    });
+
+    afterEach(function() {
+      return test.client.disconnect().then(function() {
+        test.client = undefined;
+      });
+    });
+
+    it('should connect, send, and receive a message', function(done) {
+      var msgVal = uuid.v4();
+      expect(config.senderLink, 'Required environment variables').to.exist;
+      test.client.connect(config.address)
+        .then(function() {
+          return Promise.all(_.range(config.partitionCount).
+            map(function(partition) { return test.client.createReceiver(config.receiverLinkPrefix + partition); }).
+            concat(test.client.createSender(config.senderLink)));
+        })
+        .then(function(links) {
+          var sender = links.pop();
+          _.each(links, function(receiver) {
+            receiver.on('message', function (message) {
+              expect(message).to.exist;
+              expect(message.body).to.exist;
+              // Ignore messages that aren't from us.
+              if (!!message.body.DataValue && message.body.DataValue === msgVal) {
+                done();
```

Comment:
Actually, yes it is, but I can't blame you for not knowing. In Event Hubs, each message goes to a single partition, so the only way to make sure you get the message is to a) on sending, send to a specific partition (I'll be writing that test later, once I figure out how :)), or b) attach to all partitions.